sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcsOptions.java

```diff
@@ -68,6 +70,24 @@ public interface GcsOptions extends ApplicationNameOptions, GcpOptions, Pipeline
 
   void setExecutorService(ExecutorService value);
 
+  /**
+   * The ScheduledExecutorService instance to use to create threads, can be overridden to specify a
+   * ScheduledExecutorService that is compatible with the user's environment. If unset, the default
+   * is to create an ScheduledExecutorService with a core number of threads equal to Math.max(4,
+   * Runtime.getRuntime().availableProcessors())
+   */
+  @JsonIgnore
+  @Description(
+      "The ScheduledExecutorService instance to use to create threads, can be overridden to specify "
```

Comment:
It makes a lot of sense to not declare this pipeline option in GcsOptions since the original choice of having this in GcsOptions was done a long time ago since GCS needed it but it is used in many places.

Since it is used both during pipeline construction and pipeline execution it makes sense to move it to its own options class in `org/apache/beam/sdk/options/ExecutorOptions.java` in `sdks/java/core`

Reply:
makes sense, I'll refactor this